fgpyo/sequence.py

```diff
@@ -88,9 +99,55 @@ def longest_hp_length(bases: str) -> int:
     return max_hp
 
 
-def gc_content(bases: str) -> float:
-    """Calculates the fraction of G and C bases in a sequence."""
-    if len(bases) == 0:
+def longest_dinucleotide_run_length(bases: str) -> int:
+    """Number of bases in the longest dinucleotide run in a primer.
+
+    A dinucleotide run is when two nucleotides are repeated in tandem. For example,
+    TTGG (length = 4) or AACCAACCAA (length = 10). If there are no such runs, returns 0."""
+
+    best_length: int = 0
+
+    start = 0  # the start index of the current dinucleotide run
+    while start < len(bases) - 1:
+        # get the dinuc bases
+        first_base = bases[start]
+        second_base = bases[start + 1]
```

Comment:
Wouldn't it be easier to slice here?  I.e. `dinuc = bases[start:start+2].upper()`, and then your comparisons below could be simplified to `bases[end:end+2].upper() == dinuc`?